tests/unit/Framework/TestCaseTest.php

```diff
@@ -765,6 +857,84 @@ public function testGettingNullTestResultObject(): void
         $this->assertNull($test->getTestResultObject());
     }
 
+    public function testHasFailedReturnsFalseWhenTestHasNotRunYet(): void
+    {
+        $test = new \TestWithDifferentStatuses();
+
+        $this->assertSame(BaseTestRunner::STATUS_UNKNOWN, $test->getStatus());
+        $this->assertFalse($test->hasFailed());
```

Comment:
Not sure about this method, it appears not to be used within this code base. Running

```
$ git grep -iw hasFailed
```

yields

```
src/Framework/TestCase.php:664:    public function hasFailed(): bool
```